internal/allocation/allocation.go

```diff
@@ -9,35 +9,40 @@ import (
 	"github.com/gortc/turn"
 	"github.com/pion/logging"
 	"github.com/pion/stun"
-	"github.com/pion/turn/internal/ipnet"
 	"github.com/pkg/errors"
 )
 
 // Allocation is tied to a FiveTuple and relays traffic
 // use CreateAllocation and GetAllocation to operate
 type Allocation struct {
-	RelayAddr           net.Addr
-	Protocol            Protocol
-	TurnSocket          net.PacketConn
-	RelaySocket         net.PacketConn
-	fiveTuple           *FiveTuple
-	permissionsLock     sync.RWMutex
-	permissions         map[string]*Permission
-	channelBindingsLock sync.RWMutex
-	channelBindings     []*ChannelBind
-	lifetimeTimer       *time.Timer
-	closed              chan interface{}
-	log                 logging.LeveledLogger
+	RelayAddr   net.Addr
+	Protocol    Protocol
+	TurnSocket  net.PacketConn
+	RelaySocket net.PacketConn
+	fiveTuple   *FiveTuple
+
+	permissionsLock sync.RWMutex
+	permissions     map[string]*Permission
+
+	channelBindingsLock   sync.RWMutex
+	channelNumberBindings map[turn.ChannelNumber]*ChannelBind
```

Comment:
Are we storing information twice for performance reasons?

I would like to avoid if possible, just to avoid possible bugs in the future where people modify things incorrectly.

Reply:
I would be curious how often `GetChannelByID` and `GetChannelByAddr` is called. We can pick the best storage off of that.

If they are equal maybe we can store twice, but does feel wrong. But not a huge deal!

Reply:
Sending message with channel once,  caller will call `GetChannelByID ` method  ( or callee (peer) call `GetChannelByAddr ` ) once.  I think the most important thing is that how many the binding channels for an `Allocation` will be,  100 or 1000?